crates/red_knot_python_semantic/src/stdlib.rs

```diff
@@ -33,61 +33,64 @@ impl CoreStdlibModule {
 
 /// Lookup the type of `symbol` in a given core module
 ///
-/// Returns `Unbound` if the given core module cannot be resolved for some reason
+/// Returns `SymbolLookupResult::Unbound` if the given core module cannot be resolved for some reason
 fn core_module_symbol_ty<'db>(
     db: &'db dyn Db,
     core_module: CoreStdlibModule,
     symbol: &str,
-) -> Type<'db> {
+) -> SymbolLookupResult<'db> {
     resolve_module(db, &core_module.name())
         .map(|module| global_symbol_ty(db, module.file(), symbol))
-        .map(|ty| {
-            if ty.is_unbound() {
-                ty
+        .map(|res| {
+            if res.is_unbound() {
+                res
             } else {
-                ty.replace_unbound_with(db, Type::Never)
+                res.replace_unbound_with(db, Type::Never)
```

Comment:
Given that all these methods now return `SymbolLookupResult`, it seems like we should maybe not do this here anymore? This is basically just unconditionally ignoring maybe-bound state and transforming it to always-bound. Is there any reason not to go ahead and inform the caller accurately if the name is maybe-bound? The caller can still ignore this fact if they want.

(This could be a Todo and investigated as a follow-up, if it looks like the blast radius would be large.)

Reply:
It looks like the blast radius is zero — all tests pass without this replacement logic. I was questioning why this was here and tried to reproduce it faithfully, but I fully agree with the analysis that this should be handled by the callers of this method. In fact, the callers are already forced to handle it with this new approach.

I tried (somewhat hard) to write a test case for this, but haven't succeeded so far. Postponed for now.

Reply:
I'm not worried about not having a test specifically for this.

Writing a test for it that isn't overly dependent on details of typeshed would probably require a custom typeshed, which is a feature we want to add to mdtest but haven't yet.